test/LexerTest.php

```diff
@@ -188,4 +238,47 @@ public function concatenatorProvider()
             'slash' => array(Lexer::T_SLASH),
         );
     }
+
+    /**
+     * test isOpeningQuotation
+     *
+     * @dataProvider quotationCharProvider
+     *
+     * @param string $expected      Expected outcome
+     * @param string $quotationChar quotation character (' || ")
+     *
+     * @return void
+     */
+    public function testIsOpeningQuotation($expected, $quotationChar)
+    {
+        $this->assertSame($expected, Lexer::isOpeningQuotation($quotationChar));
+    }
+
+    /**
+     * @return array
+     */
+    public function quotationCharProvider()
+    {
+        return array(
+            '1st single' => array(true, Lexer::T_SINGLE_QUOTE),
+            '2nd single' => array(false, Lexer::T_SINGLE_QUOTE),
+            '1st double' => array(true, Lexer::T_DOUBLE_QUOTE),
+            '2nd double' => array(false, Lexer::T_DOUBLE_QUOTE),
+        );
+    }
+
+    /**
+     * test set of field concatenators
+     *
+     * @dataProvider quotationCharProvider
+     *
+     * @param boolean $tmp           not used
+     * @param string  $quotationChar Character representing a string quotationChar.
+     *
+     * @return void
+     */
+    public function testIsFieldQuotationChar($tmp, $quotationChar)
```

Comment:
this is a dummy param, so I could reuse the dataProvider... but yeah, I should refactor that ..